packages/wallet/src/services/DelegationTracker/RewardsHistory.ts

```diff
@@ -0,0 +1,56 @@
+import { BigIntMath, Cardano, EpochRewards, RewardHistoryProps, WalletProvider } from '@cardano-sdk/core';
+import { KeyManagement, ProviderWithArg, Transactions } from '../..';
+import { certificateTransactions } from './util';
+import { filter, from, map, mergeMap, take } from 'rxjs';
+
+interface RewardsHistoryProviderInternals {
+  getRewardsHistory?: ProviderWithArg<EpochRewards[], RewardHistoryProps>;
+}
+
+export const getEpoch = (slotNo: number) => {
+  // Review: we should probably be getting these constants from the provider?
```

Comment:
What about if we instead just lookup the block to get the epoch? These constants are going to be awkward to maintain in our own providers, and likely impossible to get from 3rd party providers such as Blockfrost.

Reply:
Make a WalletProvider request to look up the block? That's an option, but not great for something that can be computed locally. I think something like `WalletProvider { slotEpoch(slot): Promise }` would be best, then if the provider supported something like  `{ eras: { firstEpoch; firstSlot; slotLength; }[] }` it could be implemented to compute locally.

Reply:
Yes, that design also occurred to me, but if we're calling a backend service for the constants why not just keep it consistent? This function is only being called off the back of a delegation being made right? In that case it will be very rarely called.

Reply:
I guess it boils down to saving a few requests vs a more generic WalletProvider. Can we keep this hardcoded implementation for now and do this as a separate PR?

Reply:
These constants are network-specific, so to pursue this approach we'd need some type of config to correlate the network.